packages/expo-asset/plugin/src/withAssetsAndroid.ts

```diff
@@ -0,0 +1,40 @@
+import { ConfigPlugin, withDangerousMod } from 'expo/config-plugins';
+import fs from 'fs/promises';
+import path from 'path';
+
+import { fontTypes, imageTypes, mediaTypes, resolveAssetPaths, validateAssets } from './utils';
+
+export const withAssetsAndroid: ConfigPlugin<string[]> = (config, assets) => {
+  return withDangerousMod(config, [
+    'android',
+    async (config) => {
+      const resolvedAssets = await resolveAssetPaths(assets, config.modRequest.projectRoot);
+      const validAssets = validateAssets(resolvedAssets);
+      await Promise.all(
+        validAssets.map(async (asset) => {
+          const assetsDir = getAssetDir(asset, config.modRequest.platformProjectRoot);
+          await fs.mkdir(assetsDir, { recursive: true });
```

Comment:
I think this method is well-synchronized inside of node, but there's a (good) chance that multiple assets will have the same destination base dir and this will be called with the same directory at the same time. Might be worth testing quickly to ensure there's no race conditions.

Also could just do this serially or do in two sweeps where all the directories are created first (serially) and then the files are copied (in parallel) since the file copies are the expensive bit.

Reply:
I've made the folder creation sync and kept the copy async